src/dto/EmbedPageMessage.js

```diff
@@ -0,0 +1,63 @@
+import { MessageActionRow, MessageButton } from 'discord.js';
+
+/**
+ *
+ * @type {Map<String, EmbedPageMessage>}
+ */
+const cache = new Map();
+
+export default class EmbedPageMessage {
+    /**
+     *
+     * @param { EmbedPage[] } pages
+     */
+    constructor(pages) {
+        if (pages.length === 0) {
+            throw new Error('Page Message must contain at least one page');
+        }
+        this.pages = pages;
+        this.index = 0;
+        this.message = null;
+    }
+
+    /**
+     *
+     * @param {import('discord.js').Message} message
+     */
+    async send(message = null) {
+        this.message = await message.reply('settings');
+        cache[this.message.id] = this;
+    }
+
+    async update(message) {
+        if (!this.message) {
+            await this.send(message);
+        }
```

Comment:
Continuation of some other reviews, send the message in constructor and then make update take in interaction.